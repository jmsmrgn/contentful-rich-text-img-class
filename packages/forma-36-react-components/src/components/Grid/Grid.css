@import 'settings/colors';
@import 'settings/typography';
@import 'settings/transitions';

.Grid {
  display: grid;
}

.Grid__inline {
  display: inline-grid;
}
