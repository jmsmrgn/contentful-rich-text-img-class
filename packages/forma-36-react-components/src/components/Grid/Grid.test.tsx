import React from 'react';
import { shallow, mount } from 'enzyme';
import axe from '../../utils/axeHelper';
import Grid from './Grid';

it('renders the component', () => {
  const output = shallow(<Grid>Grid</Grid>);

  expect(output).toMatchSnapshot();
});

it('renders the component with an additional class name', () => {
  const output = shallow(<Grid className="my-extra-class">Grid</Grid>);

  expect(output).toMatchSnapshot();
});

it('has no a11y issues', async () => {
  const output = mount(<Grid>Grid</Grid>).html();
  const results = await axe(output);

  expect(results).toHaveNoViolations();
});

describe('should have correct styles', () => {
  it('rowGap should be 0 by default', () => {
    const output = shallow(<Grid>Grid</Grid>);
    const outputStyles = output.get(0).props.style;
    expect(outputStyles).toMatchObject({ gridRowGap: 0 });
  });

  it('columnGap should be 1rem by default', () => {
    const output = shallow(<Grid>Grid</Grid>);
    const outputStyles = output.get(0).props.style;
    expect(outputStyles).toMatchObject({ gridColumnGap: '1rem' });
  });

  it('should has correct numbers of columns', () => {
    const output = shallow(<Grid columns={12}>Grid</Grid>);
    const outputStyles = output.get(0).props.style;
    expect(outputStyles).toMatchObject({
      gridTemplateColumns: 'repeat(12, minmax(0, 1fr)',
    });
  });

  it('should has correct numbers of rows', () => {
    const output = shallow(<Grid rows={12}>Grid</Grid>);
    const outputStyles = output.get(0).props.style;
    expect(outputStyles).toMatchObject({
      gridTemplateRows: 'repeat(12, minmax(0, 1fr)',
    });
  });
});
