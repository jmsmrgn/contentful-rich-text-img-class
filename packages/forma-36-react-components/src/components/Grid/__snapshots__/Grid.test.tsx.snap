// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component 1`] = `
<div
  className="Grid"
  data-test-id="cf-ui-grid"
  style={
    Object {
      "alignContent": undefined,
      "flow": undefined,
      "gridColumnGap": "1rem",
      "gridRowGap": 0,
      "gridTemplateColumns": "auto",
      "gridTemplateRows": "auto",
      "justifyContent": undefined,
    }
  }
>
  Grid
</div>
`;

exports[`renders the component with an additional class name 1`] = `
<div
  className="Grid my-extra-class"
  data-test-id="cf-ui-grid"
  style={
    Object {
      "alignContent": undefined,
      "flow": undefined,
      "gridColumnGap": "1rem",
      "gridRowGap": 0,
      "gridTemplateColumns": "auto",
      "gridTemplateRows": "auto",
      "justifyContent": undefined,
    }
  }
>
  Grid
</div>
`;
